v0/pipeline/steps/ci/docker-step-info.yaml

```diff
@@ -2,10 +2,16 @@ title: DockerStepInfo
 allOf:
 - $ref: ../../steps/ci/ci-step-info.yaml
 - type: object
-  required:
-  - connectorRef
-  - repo
-  - tags
+  oneOf:
+   - required:
+      - connectorRef
+      - repo
+      - tags
+   - required:
+      - registryRef
+      - useHarnessRegistry
+      - repo
+      - tags
```

Comment:
so user can configure one of: 
```
      - connectorRef
      - repo
      - tags
      ```
      or
      ```
      - registryRef
      - useHarnessRegistry
      - repo
      - tags
      ```
     what if in second case user will set **useHarnessRegistry** false?
     we can end with an inconsistent config.
     
     Why do we need **useHarnessRegistry**  at all?
    we have 
    ```
      - repo
      - tags
      ```
      which are common fields, and we have **connectorRef** and **registryRef**, which can go under **oneOf**.
      @arvind-choudhary-h , Harsh Jain, what do you think?

Reply:
to me, it's makes sense to not use `useHarnessRegistry`

Reply:
done